kectl/app/cmd/cloud/init.go

```diff
@@ -0,0 +1,163 @@
+	return cmd
+}
+
+//newInitOptions will initialise new instance of options everytime
+func newInitOptions() *options.InitOptions {
+	var opts *options.InitOptions
+	opts = &options.InitOptions{}
+	opts.DockerVersion = options.DefaultDockerVersion
+	opts.KubeedgeVersion = options.DefaultKubeEdgeVersion
+	opts.Kubernetesversion = options.DefaultK8SVersion
+
+	return opts
+}
+
+func addJoinOtherFlags(cmd *cobra.Command, initOpts *options.InitOptions) {
+
+	cmd.Flags().StringVar(&initOpts.KubeedgeVersion, options.KubeedgeVersion, initOpts.KubeedgeVersion,
+		"Use this key to download and use the required KubeEdge version")
+	cmd.Flags().StringVar(&initOpts.DockerVersion, options.DockerVersion, initOpts.DockerVersion,
+		"Use this key to download and use the required Docker version")
+	cmd.Flags().StringVar(&initOpts.Kubernetesversion, options.Kubernetesversion, initOpts.Kubernetesversion,
+		"Use this key to download and use the required Kubernetes version")
+}
+
+func Add2ToolsList(toolList map[string]util.ToolsInstallerCloud, flagData map[string]FlagData, initOptions *options.InitOptions) {
+	var kubeVer, dockerVer, k8sVer string
+
+	flgData, ok := flagData[options.KubeedgeVersion]
+	if ok {
+		fmt.Println(options.KubeedgeVersion, "VAL:", flgData.Val.(string), "DEFVAL:", flgData.DefVal.(string))
+		kubeVer = CheckIfAvailable(flgData.Val.(string), flgData.DefVal.(string))
+	} else {
+		kubeVer = initOptions.KubeedgeVersion
+	}
+	fmt.Println(options.KubeedgeVersion, "VAL:", kubeVer, "DEFVAL:", flgData.DefVal.(string))
+	toolList["Cloud"] = &util.KubeCloudInstTool{Common: util.Common{ToolVersion: kubeVer}}
+
+	flgData, ok = flagData[options.DockerVersion]
+	fmt.Println("flagdata docker is %v", flgData)
+	if ok {
+		fmt.Println(options.DockerVersion, "VAL:", flgData.Val.(string), "DEFVAL:", flgData.DefVal.(string))
+		dockerVer = CheckIfAvailable(flgData.Val.(string), flgData.DefVal.(string))
+	} else {
+		dockerVer = initOptions.DockerVersion
+	}
+	fmt.Println(options.DockerVersion, "VAL:", dockerVer, "DEFVAL:", flgData.DefVal.(string))
+	toolList["Docker"] = &util.DockerInstTool{Common: util.Common{ToolVersion: dockerVer}, DefaultToolVer: flgData.DefVal.(string)}
+
+	flgData, ok = flagData[options.Kubernetesversion]
+	if ok {
+		fmt.Println(options.Kubernetesversion, "VAL:", flgData.Val.(string), "DEFVAL:", flgData.DefVal.(string))
+		k8sVer = CheckIfAvailable(flgData.Val.(string), flgData.DefVal.(string))
+	} else {
+		k8sVer = initOptions.Kubernetesversion
+	}
+	fmt.Println(options.Kubernetesversion, "VAL:", k8sVer, "DEFVAL:", flgData.DefVal.(string))
+	toolList["Kubernetes"] = &util.K8SInstTool{Common: util.Common{ToolVersion: k8sVer}, IsEdgeNode: false, DefaultToolVer: flgData.DefVal.(string)}
+
+}
+
+func AddToolVals(f *pflag.Flag, flagData map[string]FlagData) {
+	flagData[f.Name] = FlagData{Val: f.Value.String(), DefVal: f.DefValue}
+}
+
+//Install all the required tools
+func Execute(toolList map[string]util.ToolsInstallerCloud) {
+
```

Comment:
please remove blank line at the starting and ending of functions.
Applicable to all similar places